glib_typeof relies on C++11, so we need to fall back to regular typeof
(supported by GCC and Clang) to build with older compilers.

--- gio/gcredentials.c.orig	2022-08-20 08:42:34.000000000 -0400
+++ gio/gcredentials.c	2022-08-20 08:46:40.000000000 -0400
@@ -225,7 +225,7 @@
 {
   GString *ret;
 #if G_CREDENTIALS_USE_APPLE_XUCRED
-  glib_typeof (credentials->native.cr_ngroups) i;
+  typeof (credentials->native.cr_ngroups) i;
 #endif
 
   g_return_val_if_fail (G_IS_CREDENTIALS (credentials), NULL);
